src/cclib/parser/logfileparser.py

```diff
@@ -86,20 +86,23 @@ def __init__(self, source, pos=0):
             self.src.seek(0, 2)
             self.size = self.src.tell()
             self.src.seek(pos, 0)
-            self.pos = pos
 
-        except:
+        except (AttributeError, IOError):
 
             # Stream returned by urllib should have size information.
             if hasattr(self.src, 'headers') and 'content-length' in self.src.headers:
                 self.size = int(self.src.headers['content-length'])
+            else:
+                self.size = pos
 
-            # Assume the position is what was passed to the constructor.
-            self.pos = pos
+        # Assume the position is what was passed to the constructor.
+        self.pos = pos
 
     def next(self):
         line = next(self.src)
         self.pos += len(line)
+        if self.size < self.pos:
+            self.size = self.pos
```

Comment:
Well, this shouldn't happen now since the stream is read at the start - removed. I think the next step is getting rid of FileWrapper and creating something in ccio that deals with files and streams transparently, because this is starting to get confusing!